### ISSUE TITLE
Full screen for IOS

### ISSUE THREAD

**Issue by onigetoc** [videojs/videojs-youtube#431]
Full screen do not work on IOS device.

**Comment by varemenos** [comment]
![screen shot 2017-03-06 at 12 39 50](https://cloud.githubusercontent.com/assets/626343/23606579/07ed7b30-026a-11e7-93b8-4c71b1ebe3d9.png)
![screen shot 2017-03-06 at 12 33 44](https://cloud.githubusercontent.com/assets/626343/23606454/7d1288ac-0269-11e7-8642-1d0eedde9d17.png)
![screen shot 2017-03-06 at 12 35 49](https://cloud.githubusercontent.com/assets/626343/23606455/7d12da5a-0269-11e7-86f2-87377baf51fd.png)

> undefined is not a function (evaluating 'this.tech_[method](arg)')
method is either `enterFullScreen`, `exitFullScreen`

**Comment by gkatsev** [comment]
So, looking into this, there's basically no way of actually putting the youtube video into fullscreen. Youtube doesn't have an API for it.
There are two possible solutions:
1. disable fullscreen on those devices
2. allow videojs to use "full window mode"

2 looks something like: 
![img_0001](https://cloud.githubusercontent.com/assets/535884/25876359/ab9c4da8-34ea-11e7-9e7d-2cd432219b67.png)
Basically, it'll stretch the player to be 100% in both width and height. Unless there are objections, I think we should go with 2.

Thoughts @onigetoc @varemenos?

**Comment by gkatsev** [comment]
Just realized there's a 3rd option:
3. Always use the native youtube controls on iOS.

This is because on iOS it seems to drop down to the native control bar which does have a fullscreen button.

**Comment by onigetoc** [comment]
Yes, i though to disable Videojs on IOS, but it's hard because i have one player and mp4 and other video work fullscreen except Youtube, it's will be hard but i guess not impossible with the same player to go back to videojs play to native youtube player with the same player.

**Comment by gkatsev** [comment]
There's no need to disable videojs. You can just tell videojs to use the native controls instead of using its custom controls.

So, to re-iterate, the 3 options are:
1. disable fullscreen
2. use "full window" mode
3. disable the custom controls and use native iOS controls by default

**Comment by onigetoc** [comment]
Ok, but for now, i do not how to do it with javascript to diseable the controler and enable it again with the same player based on a Youtube video or not.

**Comment by gkatsev** [comment]
Ah, I see. I'm not 100% sure, I'll get back to you on that.

**Comment by onigetoc** [comment]
I did a example here. Mixing Youtube video with MP4 video
https://jsfiddle.net/onigetoc/ytntz2ux/

**Comment by varemenos** [comment]
I'm not entirely sure which is the right option to choose.

Personal, I wrote a `supportsFullscreen()` function that detects whether the Fullscreen API is supported or not in the current device which toggles the visibility of the control bar's fullscreen button.

```js
var supportsFullscreen = function () {
    if (document.cancelFullScreen || document.mozCancelFullScreen || document.webkitCancelFullScreen) {
        return true;
    } else {
        return false;
    }
};
```

Below is part of the video.js configuration that utilizes the supportsFullscreen function.

```js
var videojsPlayer = videojs(
    'my-player',
    {
        controlBar: {
            fullscreenToggle: supportsFullscreen()
        }
    }
);
```
